### DISCUSSION TITLE
Push declined due to email privacy restrictions

### DISCUSSION THREAD

**Discussion by tothomas69** [community/community#56269] [New to GitHub]
New to all of this and not afraid to read up. I am getting this error when trying to push to a repository. The blogs and searches revealed how to set this up and why. I did those steps and there is no change to the behavior. 30668901+tothomas69@users.noreply.github.com is the email address I have set in gitkraken, I got that email address from github settings email. 

I have also searched here for this issue, did not find anything actionable.

**Comment by airtower-luna** [comment]
Most likely one of the addresses you want to keep private is still recorded in one of the commits as the author or committer address. If that's the case, you'll need to [edit the repository history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) to replace it. If there are a lot of commits like that, take a look at [`git-filter-repo`](https://github.com/newren/git-filter-repo).
